### PR TITLE
Bug 1516475 - Support for Thunderbird MSI installers

### PR DESCRIPTION
**Pull request by jfx2006**
Adding support for Thunderbird MSIs

### CODE REVIEW

File: bouncerscript/constants.py [hunk 1]
```diff
@@ -34,7 +37,7 @@
     'firefox-rc': r'^(/firefox/candidates/.*?/build[0-9]+/(update/)?(?:linux-i686|linux-x86_64|mac|win32|win64(?:|-aarch64))/\:lang/(?:firefox|Firefox).*\.(?:bz2|dmg|exe|mar))$',
     'firefox': r'^(/firefox/releases/.*?/(update/)?(?:linux-i686|linux-x86_64|mac|win32|win64(?:|-aarch64))/\:lang/(?:firefox|Firefox).*\.(?:bz2|dmg|exe|mar|msi))$',
     'devedition': r'^(/devedition/releases/.*?/(update/)?(?:linux-i686|linux-x86_64|mac|win32|win64(?:|-aarch64))/\:lang/(?:firefox|Firefox).*\.(?:bz2|dmg|exe|mar|msi))$',
-    'thunderbird': r'^(/thunderbird/releases/.*?/(update/)?(?:linux-i686|linux-x86_64|mac|win32|win64(?:|-aarch64))/\:lang/(?:thunderbird|Thunderbird).*\.(?:bz2|dmg|exe|mar))$',
+    'thunderbird': r'^(/thunderbird/releases/.*?/(update/)?(?:linux-i686|linux-x86_64|mac|win32|win64(?:|-aarch64))/\:lang/(?:thunderbird|Thunderbird).*\.(?:bz2|dmg|exe|mar|msi))$',
```
Comment (Callek) [inline bouncerscript/constants.py]: ./bouncerscript/constants.py:40:181: E501 line too long (181 > 180 characters)
    'thunderbird': r'^(/thunderbird/releases/.*?/(update/)?(?:linux-i686|linux-x86_64|mac|win32|win64(?:|-aarch64))/\:lang/(?:thunderbird|Thunderbird).*\.(?:bz2|dmg|exe|mar|msi))$',

File: bouncerscript/constants.py [hunk 2]
```diff
@@ -1,10 +1,13 @@
 ALIASES_REGEXES = {
     'thunderbird-beta-latest': r'^Thunderbird-\d+\.0b\d+$',
     'thunderbird-beta-latest-ssl': r'^Thunderbird-\d+\.0b\d+-SSL$',
+    'thunderbird-beta-msi-latest-ssl': r'^Thunderbird-\d+\.0b\d+-msi-SSL$',
     'thunderbird-next-latest': r'^Thunderbird-\d+\.\d+(\.\d+)?$',
     'thunderbird-next-latest-ssl': r'^Thunderbird-\d+\.\d+(\.\d+)?-SSL$',
+    'thunderbird-next-msi-latest-ssl': r'^Thunderbird-\d+\.\d+(\.\d+)?-msi-SSL$',
```
Comment (MihaiTabara) [inline bouncerscript/constants.py]: Hm, I wonder if we actually need this?

File: bouncerscript/constants.py [hunk 3]
```diff
@@ -34,7 +37,8 @@
     'firefox-rc': r'^(/firefox/candidates/.*?/build[0-9]+/(update/)?(?:linux-i686|linux-x86_64|mac|win32|win64(?:|-aarch64))/\:lang/(?:firefox|Firefox).*\.(?:bz2|dmg|exe|mar))$',
     'firefox': r'^(/firefox/releases/.*?/(update/)?(?:linux-i686|linux-x86_64|mac|win32|win64(?:|-aarch64))/\:lang/(?:firefox|Firefox).*\.(?:bz2|dmg|exe|mar|msi))$',
     'devedition': r'^(/devedition/releases/.*?/(update/)?(?:linux-i686|linux-x86_64|mac|win32|win64(?:|-aarch64))/\:lang/(?:firefox|Firefox).*\.(?:bz2|dmg|exe|mar|msi))$',
-    'thunderbird': r'^(/thunderbird/releases/.*?/(update/)?(?:linux-i686|linux-x86_64|mac|win32|win64(?:|-aarch64))/\:lang/(?:thunderbird|Thunderbird).*\.(?:bz2|dmg|exe|mar))$',
+    'thunderbird': (r'^(/thunderbird/releases/.*?/(update/)?(?:linux-i686|linux-x86_64|mac|win32|win64(?:|-aarch64))/\:lang/'
+                    r'(?:thunderbird|Thunderbird).*\.(?:bz2|dmg|exe|mar|msi))$'),
```
Comment (MihaiTabara) [inline bouncerscript/constants.py]: Nit: ideally we add some tests as well to make sure the regex validates before we let this go in production.

Comment (MihaiTabara) [reply bouncerscript/constants.py]: Thanks for adding the tests!

**Comment by MihaiTabara** [comment]
Rolled-out to production via https://github.com/mozilla-releng/build-puppet/pull/467

**Pull request merged**